src/compiler/program.ts

```diff
@@ -3760,7 +3760,7 @@ namespace ts {
         }
 
         function handleDirectoryCouldBeSymlink(directory: string) {
-            if (!host.getResolvedProjectReferences()) return;
+            if (!host.getResolvedProjectReferences() || pathContainsPnpmDirectory(directory)) return;

```

Comment:
This doesnt look right? If the symlink is in .pnpm for referenced project file, it still should be able to handle the symlink from program creation perspective?

Reply:
This is just a slight optimization since my change to `setSymlinkedDirectory` will end up making this function a no-op. So we can continue that conversation in your other comment.